neo/Network/LocalNode.cs

```diff
@@ -136,17 +142,42 @@ private void AddTransactionLoop()
             while (!cancellationTokenSource.IsCancellationRequested)
             {
                 new_tx_event.WaitOne();
+
+                bool shouldCoolDown = false;
+                MemPoolReadWriteLock.EnterReadLock();
+                if (mem_pool.Count > 1000 && temp_pool.Count > 1000) shouldCoolDown = true;
+                MemPoolReadWriteLock.ExitReadLock();
+                // Don't starve readers wanting to acquire the mem_pool lock
+                if (shouldCoolDown) Thread.Sleep(1000);
+                
                 Transaction[] transactions;
                 lock (temp_pool)
                 {
                     if (temp_pool.Count == 0) continue;
-                    transactions = temp_pool.ToArray();
-                    temp_pool.Clear();
+                    if (temp_pool.Count < 1000)
```

Comment:
I should make the 1000 here a constant to make it more easily configurable

Reply:
would be a bit more accurate actually if the condition were <= instead of <, but it doesn't make much of a difference just optimizes for the case that temp_pool.Count is exactly 1000.